### PR TITLE
Copy user fields

### PR DESCRIPTION
**Pull request by mireaulf**
Some stylish changes to be more pythonic

### CODE REVIEW

File: copy_user_fields.py [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+            if not (previous['fieldType'] == fields[i]['fieldType'] and
+                    previous['contentType'] == fields[i]['contentType'] and
+                    previous['fieldQueries'] == fields[i]['fieldQueries'] and
+                    previous['freeTextQueries'] == fields[i]['freeTextQueries'] and
+                    previous['facet'] == fields[i]['facet'] and
+                    previous['multivalueFacet'] == fields[i]['multivalueFacet'] and
+                    previous['sort'] == fields[i]['sort'] and
+                    previous['displayField'] == fields[i]['displayField']):
+                print(
+                    f'SKIPPING FIELD {current["name"]}. Found fields in CloudV1 with the same name but different configurations: {fields}')
```
Comment (malaporte) [inline copy_user_fields.py:34]: Maybe we can be a bit more tolerant here... for example, most flags could be set as soon as ONE instance has it in V1.

**Review by malaporte — APPROVED** [review]

File: copy_user_fields.py (hunk 1 above)
Comment (mireaulf) [reply copy_user_fields.py:34]: So if we have 3 fields with the same name (on 3 different sources) in V1, as soon as 1 of the flags is true the new V2 field would have it set to true?

Comment (malaporte) [reply copy_user_fields.py:34]: Yes, I think that's the best we can do...

File: client/cloud_v1.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+    def sources_get(self):
+        return self.do_get(f'rest/workgroups/{self.org_id}/sources')
+
+    def fields_get(self) -> list:
+        return self.do_get(f'rest/workgroups/{self.org_id}/fields')
+
+    def fields_get_for_source(self, source_id):
+        fields = self.fields_get()
+        return [field for field in fields if field["sourceId"] == source_id]
```
Comment (benjamin-ross) [inline client/cloud_v1.py:32]: Good old filter, you can also do a filter here: 
filter(lambda field: field["sourceId"] == source_id, fields)

File: copy_user_fields.py [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+        # deduplicate
+        return dict([(field[0],
+                      [f[1] for f in v1_fields_by_name if f[0] == field[0]])
+                     for field in v1_fields_by_name])
+
+    def validate_same_config(fields: list) -> bool:
+        previous = fields[0]
+        for i in range(1, len(fields)):
+            current = fields[i]
+            if not (previous['fieldType'] == fields[i]['fieldType'] and
```
Comment (benjamin-ross) [inline copy_user_fields.py:25]: Can you move this condition in a function/lambda ?

File: copy_user_fields.py [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+            previous = fields[i]
+        return True
+
+    fields_by_name = get_fields_by_name()
+    unique_fields_by_name = \
+        [t for t in
+         [(field, fields_by_name[field])
+          if validate_same_config(fields_by_name[field]) else None
+          for field in fields_by_name]
+         if t is not None]
```
Comment (benjamin-ross) [inline copy_user_fields.py:45]: Why not just do a empty list with a unique for that append to it (like in line #56)?

File: copy_user_fields.py [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+
+from client.cloud_v2 import *
+from client.cloud_v1 import *
+from client.fields import Fields
+
+
+def v1_get_unique_fields(fields: list) -> list:
+    def get_fields_by_name() -> dict:
+        v1_fields_by_name = [(field['name'], field) for field in fields]
+        # deduplicate
```
Comment (benjamin-ross) [inline copy_user_fields.py:16]: A lot of gymnastics for 
```
d = dict()
for field in fields:
	name = field['name']	
	if name in d:
		d[name].append(field) 
	else:
		d[name] = [field]
return d
```

File: copy_user_fields.py (hunk 1 above)
Comment (mireaulf) [reply copy_user_fields.py:34]: I'll still validate that the fields have the same fieldType and the contentType.

File: copy_user_fields.py (hunk 5 above)
Comment (mireaulf) [reply copy_user_fields.py:16]: I guess that's what happens when you work on the same thing for too long 😄

**Review by benjamin-ross — APPROVED** [review]

**Comment by mireaulf** [comment]
I will merge this PR but still make the changes in another one.

**Pull request merged**